L50-objc-interceptor.md

```diff
@@ -0,0 +1,263 @@
+gRPC Objective-C Interceptor
+----
+* Author(s): mxyan
+* Approver: psrini
+* Status: In Review
+* Implemented in: Objective-C
+* Last updated: 2019-04-18
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/7QGmsQW86Ss
+
+## Abstract
+
+Proposes the interceptor architecture and API for gRPC Objective-C.
+
+## Background
+
+gRPC interceptor has been a popular utility for users making HTTP requests.
+gRPC Objective-C library has received a number of requests to add support of
+this feature to gRPC. The old gRPC Objective-C library API was not flexiable enough to support interceptor API. With the [new API](https://github.com/grpc/proposal/blob/master/L38-objc-api-upgrade.md), now it is possible to add in this support.
+
+### Use cases
+The use cases for interceptor is wide; a few examples include:
+* Log RPCs
+* Collect metrics
+* Obtain/Refresh access tokens
+* Create test stub
+* ...
```

Comment:
Caching should be  a popular one, especially when you already have an example on it.